File: 0001-0500/0024/0024.swap-nodes-in-pairs.js
/*
 * @lc app=leetcode id=24 lang=javascript
 *
 * [24] Swap Nodes in Pairs
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var swapPairs = function (head) {
    let current = head;
    while (current && current.next) {
        let temp = current.val;
        current.val = current.next.val;
        current.next.val = temp;
        current = current.next.next;
    }
    return head;
};
// @lc code=end

File: 0001-0500/0024/0024.swap-nodes-in-pairs.cs
/*
 * @lc app=leetcode id=24 lang=csharp
 *
 * [24] Swap Nodes in Pairs
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode SwapPairs(ListNode head) {
        if(head == null || head.next == null) return head;
        ListNode next = head.next;
        head.next = SwapPairs(next.next);
        next.next = head;
        return next;
    }
}
// @lc code=end

